File: oesp_sdk_python/oesp_sdk/sync/env.py
import os
from typing import Optional, TypedDict

class SyncConfig(TypedDict):
    base_url: str
    api_key: Optional[str]
    timeout_sec: float
    max_chunk_bytes: int

def get_sync_config(
    base_url: Optional[str] = None,
    api_key: Optional[str] = None,
    timeout_sec: float = 30.0,
    max_chunk_bytes: int = 500000
) -> SyncConfig:
    default_base_url = "http://oesp-sync-server:8000"
    env_base_url = os.environ.get("OESP_SYNC_BASE_URL")
    
    return {
        "base_url": base_url or env_base_url or default_base_url,
        "api_key": api_key,
        "timeout_sec": timeout_sec,
        "max_chunk_bytes": max_chunk_bytes
    }

File: oesp-sdk/packages/sync-http/src/env.ts
export interface SyncConfig {
  baseUrl: string;
  apiKey?: string;
  timeoutMs: number;
  maxChunkBytes: number;
}

export function getSyncConfig(overrides: Partial<SyncConfig> = {}): SyncConfig {
  const defaultBaseUrl = "http://oesp-sync-server:8000";
  
  let envBaseUrl: string | undefined;
  
  // Node.js
  const nodeEnv = (globalThis as any)?.process?.env;
  if (nodeEnv?.OESP_SYNC_BASE_URL) {
    envBaseUrl = nodeEnv.OESP_SYNC_BASE_URL;
  }

  
  // Vite / Web
  // @ts-ignore
  if (typeof import.meta !== "undefined" && import.meta.env?.VITE_OESP_SYNC_BASE_URL) {
    // @ts-ignore
    envBaseUrl = import.meta.env.VITE_OESP_SYNC_BASE_URL;
  }

  return {
    baseUrl: overrides.baseUrl || envBaseUrl || defaultBaseUrl,
    apiKey: overrides.apiKey,
    timeoutMs: overrides.timeoutMs || 30000,
    maxChunkBytes: overrides.maxChunkBytes || 500000, // 500KB
  };
}
